src/utils.jl

```diff
@@ -0,0 +1,126 @@
+function _prepare_cohort_directory(
+    path::String
+)
+    mkpath(path)
+    return path
+end
+
+function _load_metadata(
+    metadata_path::String,
+    metadata_check::Bool
+)
+    if metadata_check && isfile(metadata_path)
+        return JSON3.read(read(metadata_path, String), Dict{String, Any})
+    end
+    return Dict{String, Any}()
+end
```

Comment:
can you eliminate this funciton and put this code directly into the main funciton you defined?